metadata-ingestion/src/datahub/sql_parsing/tool_meta_extractor.py

```diff
@@ -42,14 +51,68 @@ class ToolMetaExtractor:
     by warehouse query logs.
     """
 
-    def __init__(self) -> None:
-        self.report = ToolMetaExtractorReport()
+    def __init__(
+        self,
+        report: ToolMetaExtractorReport,
+        looker_user_mapping: Optional[Dict[str, str]] = None,
+    ) -> None:
+        self.report = report
         self.known_tool_extractors: List[Tuple[str, Callable[[QueryLog], bool]]] = [
             (
                 "mode",
                 self._extract_mode_query,
-            )
+            ),
+            (
+                "looker",
+                self._extract_looker_query,
+            ),
         ]
+        # maps user id (as string) to email address
+        self.looker_user_mapping = looker_user_mapping
+
+    @classmethod
+    def create(
+        cls,
+        graph: Optional[DataHubGraph] = None,
+    ) -> "ToolMetaExtractor":
+        report = ToolMetaExtractorReport()
+        looker_user_mapping = None
+        if graph:
+            query = (
+                ElasticPlatformResourceQuery.create_from()
+                .group(LogicalOperator.AND)
+                .add_field_match(PlatformResourceSearchFields.PLATFORM, "looker")
+                .add_field_match(
+                    PlatformResourceSearchFields.RESOURCE_TYPE,
+                    "USER_ID_MAPPING",
+                )
+                .end()
+            )
+
+            platform_resources = [
+                platform_resource
+                for platform_resource in PlatformResource.search_by_filters(
+                    query=query, graph_client=graph
+                )
+            ]
+
+            if len(platform_resources) > 1:
+                report.failures.append(
+                    "Looker user metadata extraction failed. Found more than one looker user id mappings."
+                )
+            else:
+                platform_resource = platform_resources[0]
+
+                if (
+                    platform_resource
+                    and platform_resource.resource_info
+                    and platform_resource.resource_info.value
+                ):
+                    with contextlib.suppress(ValueError, AssertionError):
+                        value = platform_resource.resource_info.value.as_raw_json()
+                        if value:
+                            looker_user_mapping = value
```

Comment:
Let's extract this into method `extract_looker_user_mapping_from_graph and add explicit try..except around to  call to `extract_looker_user_mapping_from_graph` for not to break rest of connector.